src/Features/LanguageServer/ProtocolUnitTests/Definitions/GoToDefinitionTests.cs

```diff
@@ -103,7 +103,10 @@ void M()
         }
 
         private static async Task<LSP.Location[]> RunGotoDefinitionAsync(Solution solution, LSP.Location caret)
-            => await GetLanguageServer(solution).ExecuteRequestAsync<LSP.TextDocumentPositionParams, LSP.Location[]>(LSP.Methods.TextDocumentDefinitionName,
-                CreateTextDocumentPositionParams(caret), new LSP.ClientCapabilities(), null, CancellationToken.None);
+        {
+            var queue = GetRequestQueue(solution);
```

Comment:
Is there a way to avoid each test having to duplicate this? Should it just be calling into the underlying handler directly rather than having to make a queue that won't really do anything? Or is this a "feature" in that it ensures the test queueing isn't broken in some way that would prevent stuff working?

Reply:
I think its a small feature to help people write successful tests. Once a request handler is changed to use `context.Document` for example, then there needs to be something that ensures the RequestContext passed in is set up correctly by calling into the handler to get the document, etc. Either each test, or the base class, needs to do that work, or the tests can just ask the base class for a queue.